### ISSUE TITLE
Add native Informix MULTISET and nested ROW support

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#13400]
Informix has native `MULTISET` support, pretty much as specified in the SQL standard.

The following JDBC example shows how deserialising it could work:

```java
try (Statement s = connection.createStatement();
    ResultSet rs = s.executeQuery("""
        select
          first_name,
          last_name,
          multiset (select id, title from t_book where t_book.author_id = t_author.id order by id desc)
        from t_author;
        """)
) {
    while (rs.next()) {
        System.out.println(rs.getString(1) + " " + rs.getString(2));
        Object o = rs.getObject(3);

        if (o instanceof List<?> l) {
            for (Object e : l) {
                if (e instanceof Struct x) {
                    for (Object attr : x.getAttributes()) {
                        System.out.println("  " + attr + ": " + attr.getClass());
                    }
                }
            }
        }
    }
}
```

The result is:

```
George Orwell
  1: class java.lang.Integer
  1984: class java.lang.String
  2: class java.lang.Integer
  Animal Farm: class java.lang.String
Paulo Coelho
  3: class java.lang.Integer
  O Alquimista: class java.lang.String
  4: class java.lang.Integer
  Brida: class java.lang.String
```

So, we don't really have to emulate anything in Informix

### Tasks

- [x] Support nested `RowAsField`
- [x] Support nested `TableAsField`
- [x] Support `Multiset`

These aren't possible to support natively in Informix:

- Support `MultisetAgg`

### Caveats

It seems Informix has some bugs in this area, e.g.

- Some syntaxes produce internal errors, e.g. 
  - Nesting `ROW(MULTISET(...))`: https://stackoverflow.com/q/71706158/521799
  - Using `ORDER BY` in a derived table, prior to `LATERAL` unnesting: https://stackoverflow.com/q/71706335/521799
- Some syntaxes produce wrong data, to be investigated
  - Using `ORDER BY` again: https://stackoverflow.com/q/71706642/521799

I've created a new issue for this: https://github.com/jOOQ/jOOQ/issues/13402

**Comment by lukaseder** [comment]
All of this works extremely well, which is very exciting! The exception (so far) is I don't really know how to emulate `MULTISET_AGG`, which is a jOOQ-synthetic construct that can be made to work with `XMLAGG` or `JSON_ARRAYAGG`, but not really with native `MULTISET`

**Comment by lukaseder** [comment]
Some tests for deeply nested `MULTISET` and `ROW` combinations are failing:

- Due to some internal errors (probably bugs in Informix, e.g. https://stackoverflow.com/q/71706158/521799)
- Producing wrong results (probably other bugs in Informix)

Will investigate this further

**Comment by lukaseder** [comment]
Implemented in jOOQ 3.17.0. Known issues are listed in #13402, and might be addressed in the future.
